README.md

```diff
@@ -1,75 +1,93 @@
+| `server-public-key`        | true        | Path to remote server public ssl key (local path like c:\folder)                  |
+
+## Prerequisites
+
+Inbound secure WinRm network traffic (TCP port 5986) must be allowed from the GitHub Actions Runners virtual network so that remote sessions can be received.
+
+Prep the remote Windows server to accept WinRM management calls.  In general the Windows server needs to have a Web Services for Management, [WSMan], listener that looks for incoming Windows Remote Management, [WinRM], calls. Firewall exceptions need to be added for the secure WinRM TCP ports, and non-secure firewall rules should be disabled. Here is an example script that would be run on the Windows server:
+
+  ```powershell
+  $Cert = New-SelfSignedCertificate -CertstoreLocation Cert:\LocalMachine\My -DnsName <<ip-address|fqdn-host-name>>
+
+  Export-Certificate -Cert $Cert -FilePath C:\temp\<<cert-name>>
+
+  Enable-PSRemoting -SkipNetworkProfileCheck -Force
+
+  # Check for HTTP listeners
+  dir wsman:\localhost\listener
+
+  # If HTTP Listeners exist, remove them
+  Get-ChildItem WSMan:\Localhost\listener | Where -Property Keys -eq "Transport=HTTP" | Remove-Item -Recurse
+
+  # If HTTPs Listeners don't exist, add one
+  New-Item -Path WSMan:\LocalHost\Listener -Transport HTTPS -Address * -CertificateThumbPrint $Cert.Thumbprint –Force
+
+  # This allows old WinRm hosts to use port 443
+  Set-Item WSMan:\localhost\Service\EnableCompatibilityHttpsListener -Value true
+
+  # Make sure an HTTPs inbound rule is allowed
+  New-NetFirewallRule -DisplayName "Windows Remote Management (HTTPS-In)" -Name "Windows Remote Management (HTTPS-In)" -Profile Any -LocalPort 5986 -Protocol TCP
+
+  # For security reasons, you might want to disable the firewall rule for HTTP that *Enable-PSRemoting* added:
+  Disable-NetFirewallRule -DisplayName "Windows Remote Management (HTTP-In)"
+  ```
+
+- `ip-address` or `fqdn-host-name` can be used for the `DnsName` property in the certificate creation. It should be the name that the actions runner will use to call to the Windows server.
+- `cert-name` can be any name.  This file will used to secure the traffic between the actions runner and the Windows server
 
 ## Example
 
 ```yml
-# TODO: Fill in the correct usage
+...
+env:
+  cert-path: ./public-server-key
+  web-app-archive: ./src/build/web-app.zip
+
 jobs:
-  job1:
-    runs-on: [self-hosted]
+  ...
+
+  build-and-deploy:
+    runs-on: [windows-2019]
     steps:
-      - uses: actions/checkout@v2
+      # build the application files and package in a single archive file
+      build-package:
+      ...
+        with:
+          target-archive-name: ${{env.web-app-archive}}
+      ...
 
-      - name: Add the action here
-        uses: im-open/this-repo@v1.0.0
+      deploy-package:
+      uses: 'im-open/deploy-on-prem-web-app@v1.0.0'
         with:
-          input-1: 'abc'
-          input-2: '123
+          server: ${{ env.server }}
```

Comment:
You don't have to but it might be nice to add the server up in your env list above.

Reply:
👍🏽